src/go/rpk/pkg/cli/cmd/redpanda/config.go

```diff
@@ -167,6 +169,7 @@ func initNode(fs afero.Fs) *cobra.Command {
 			p := config.ParamsFromCommand(cmd)
 			cfg, err := p.Load(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
+			cfg = cfg.FileOrDefaults()
```

Comment:
Inline comment, `// we modify the node ID in the raw file without writing env / flag overrides`